client/allocrunner/group_service_hook_test.go

```diff
@@ -293,3 +294,105 @@ func TestGroupServiceHook_getWorkloadServices(t *testing.T) {
 	services := h.getWorkloadServicesLocked()
 	must.Len(t, 1, services.Services)
 }
+
+func TestGroupServiceHook_PreKill(t *testing.T) {
```

Comment:
These are great and seeing we have the pattern, would it be worth adding another test to cover the hook waiting for shutdown delay?